probe-rs/src/architecture/arm/core/m4.rs

```diff
@@ -478,6 +481,15 @@ impl<'probe> CoreInterface for M4<'probe> {
     }
 
     fn step(&mut self) -> Result<CoreInformation, Error> {
+        //First check if we stopped on a breakpoint, because this requires special handling before we can continue
+        let was_breakpoint =
+            if self.state.current_state == CoreStatus::Halted(HaltReason::Breakpoint) {
+                self.enable_breakpoints(false)?;
```

Comment:
@Tiwalun ... I agree, this doesn't allow for the case of software breakpoints.  

I am not sure I understand `setting the program counter to the next instruction` ... would that not bypass the current instruction?

Reply:
You're right, with HW breakpoints this would probably bypass the current instruction. 

In the case of SW breakpoints, this is exactly what you want, because the SW breakpoint is an instruction as well. 

This means that we  have to check if we hit a SW or HW breakpoint, and react differently. This is probably a bigger change, and should be done for all core types. I would suggest we open an issue for this, and handle it separately, what do you think? If we handle it separately, this can be merged as is, it's already an improvemen over the current statet.

Reply:
@Tiwalun ... Cool, I agree. The debugger will eventually need to set its own software breakpoints if we want to implement stepping at line level, etc. , so handling SW breakpoints is a good thing to do. I will create an Issue on the master branch for this.

Reply:
Great, thanks.